client/src/stores/NetworkStore.js

```diff
@@ -30,11 +32,17 @@ var NetworkStore = Reflux.createStore({
   onConnect: function(host, username) {
     logger.debug("Connect to " + host + " as " + username);
     this.orbit.connect(username)
-      .catch((e) => logger.error(e))
+      .catch((e) => {
+        logger.error(e)
+        // to properly stop partially executed stuff :
+        NetworkActions.disconnect();
+      })
   },
   onDisconnect: function() {
-    logger.debug("Disconnect");
+    logger.debug("Disconnect from network");
+    this.orbit.disconnect();
     this._updateNetwork(null);
+    IpfsDaemonActions.stop();
   }
```

Comment:
Let's move the stopping of daemon to IpfsDaemonStore: listen for 'disconnect' action in that store and stop the daemon.